File: apps/pipeline/fix_covers.py
import os
import sys
import asyncio
from prisma import Prisma
import random

async def main():
    db = Prisma()
    await db.connect()
    
    # 查找所有来源是 方程式新闻/BWEnews 的文章
    articles = await db.article.find_many(
        where={'source': {'contains': 'BWEnews'}}
    )
    print(f"Found {len(articles)} BWEnews articles.")
    
    for article in articles:
        seed = random.randint(1, 100000)
        new_url = f"https://loremflickr.com/1200/630/finance,crypto,business/all?lock={seed}"
        await db.article.update(
            where={'id': article.id},
            data={'cover_image': new_url}
        )
        print(f"Updated article {article.id} cover -> {new_url}")
        
    await db.disconnect()

if __name__ == "__main__":
    asyncio.run(main())

File: packages/database/fix_covers.js
const { Client } = require('pg');

async function main() {
  const client = new Client({ connectionString: 'postgresql://yayanews:<REDACTED>@127.0.0.1:5432/yayanews' });
  await client.connect();
  
  const res = await client.query(`SELECT id FROM articles WHERE source LIKE '%BWEnews%'`);
  console.log(`Found ${res.rowCount} BWEnews articles`);
  
  for (const row of res.rows) {
    const seed = Math.floor(Math.random() * 100000);
    const new_url = `https://loremflickr.com/1200/630/finance,crypto,business/all?lock=${seed}`;
    await client.query(`UPDATE articles SET cover_image = $1 WHERE id = $2`, [new_url, row.id]);
    console.log(`Updated article ${row.id} cover -> ${new_url}`);
  }
  
  await client.end();
}

main().catch(console.error);
